[a11y] Add drop zone component a11y docs and edit description (#2243)
Shopify/polaris-react: src/components/DropZone/README.md

- To provide context to upload errors when they occur, use the [banner component](https://polaris.shopify.com/components/feedback-indicators/banner)
- To provide feedback during file upload, use the [spinner component](https://polaris.shopify.com/components/feedback-indicators/spinner)

---

## Accessibility

<!-- content-for: android -->

See Material Design and development documentation about accessibility for Android:

- [Accessible design on Android](https://material.io/design/usability/accessibility.html)
- [Accessible development on Android](https://developer.android.com/guide/topics/ui/accessibility/)

<!-- /content-for -->

<!-- content-for: ios -->

See Apple’s Human Interface Guidelines and API documentation about accessibility for iOS:

- [Accessible design on iOS](https://developer.apple.com/design/human-interface-guidelines/ios/app-architecture/accessibility/)
- [Accessible development on iOS](https://developer.apple.com/accessibility/ios/)

<!-- /content-for -->

<!-- content-for: web -->

> The drop zone component builds on the native HTML `<input type=”upload”>` element by including a `<button>` as well as a drag and drop area.

Reviewer: We made a few tweaks to this component recent (which I should have caught in docs earlier!). The separate button is still there, but it's really just for looks and to give an obvious click/tap target for mouse/touch users.

```suggestion
The drop zone component builds on the native HTML `<input type="upload">` element. It includes a  visual`<button>` as well as a drag and drop area that can receive keyboard focus.
```

### Keyboard support

> To upload a file with the keyboard, merchants interact with the “Add file” button.

Reviewer: ```suggestion
To upload a file with the keyboard, merchants can interact with the drag-and-drop region.
```

> - To give the button keyboard focus, use the <kbd>tab</kbd> key (or <kbd>shift</kbd> + <kbd>tab</kbd> when tabbing backwards)

Reviewer: ```suggestion
- To give the input keyboard focus, use the <kbd>tab</kbd> key (or <kbd>shift</kbd> + <kbd>tab</kbd> when tabbing backwards)
```

> - To activate the button, use the <kbd>enter</kbd>/<kbd>return</kbd> or <kbd>space</kbd> keys

Reviewer: ```suggestion
- To activate the input, use the <kbd>enter</kbd>/<kbd>return</kbd> or <kbd>space</kbd> keys
```
